wolfcrypt/src/sha.c

```diff
@@ -385,6 +401,9 @@ int wc_ShaUpdate(Sha* sha, const byte* data, word32 len)
 
 int wc_ShaFinal(Sha* sha, byte* hash)
 {
+    if (sha == NULL || hash == NULL) {
+        return BAD_FUNC_ARG;
+    }
     byte* local = (byte*)sha->buffer;
```

Comment:
move variable declarations to beginning of statement or change to "declaration then assignement"